src/App.js

```diff
@@ -1,25 +1,145 @@
+        setDisplayValue(operation.operand1 / 100)
+      }
+    }
+
+    else if (event.target.value === "+/-") {
+
+      if (operation.operand1 && operation.operator) {
+        setOperation(prevOperation => ({ ...prevOperation, operand2: prevOperation.operand2 * -1 }))
+        setDisplayValue(operation.operand2 * -1)
+      }
+      else {
+        setOperation(prevOperation => ({ ...prevOperation, operand1: prevOperation.operand1 * -1 }))
+        setDisplayValue(operation.operand1 * -1)
+      }
+    }
+
+    else if (event.target.value === '0') {
+      if (results !== "") {
+        clear()
+        setOperation({ operand1: "", operator: null, operand2: "" })
+        selectOperand1(event.target.value)
+
+      }
+      else if (operation.operand1 === "") setDisplayValue("")
+
+      else {
+        if (operation.operand2) {
+          setOperation((prevOperation => ({ ...prevOperation, operand2: prevOperation.operand2 + event.target.value })))
+          setDisplayValue(operation.operand2 + event.target.value)
+        }
+        else {
+          setOperation((prevOperation => ({ ...prevOperation, operand1: prevOperation.operand1 + event.target.value })))
+          setDisplayValue(operation.operand1 + event.target.value)
+        }
+      }
+    }
+
+    else {
+      if (results !== "") {
+        clear()
+        setOperation({ operand1: "", operator: null, operand2: "" })
+        selectOperand1(event.target.value)
+
+      }
+      else if (operation.operand1 && operation.operator) selectOperand2(event.target.value)
+
+      else selectOperand1(event.target.value)
+
+      handleChange(event.target.value)
+    }
+  }
+
+
+
+
+
+  const calculate = (num1, num2, operator) => {
+    switch (operator) {
+      case '+': return num1 + num2
+      case '-': return num1 - num2
+      case '*': return num1 * num2
+      case '/': return num1 / num2
+      default: return 0
+
+    }
+  }
 
-function App() {
   return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
+    <div className="w-[300px] m-32">
+      <input onChange={handleChange} placeholder="" value={displayValue} name="number" className="w-[300px] outline-none bg-gray-600 p-5 text-right placeholder:text-right placeholder:text-white  text-white" />
+      <div className="grid grid-cols-4">
+        {values.map((item, index) => {
+          return (
+            <Button key={index} value={item} handleClick={handleClick} style={`${index === 3 || index === 7 || index === 11 || index === 15 || index === 18 ? "bg-orange-500" : ""} ${index === 16 ? "col-span-2" : ""}`} />
```

Comment:
To improve readability, I would suggest using conditional statements only when there are few conditions. In cases where there are many conditions, it would be better to use arrays of numbers and operators separately to map accordingly or to create an object for styles and indexes respectively.